File: liczbyp.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char **argv) {
  
    signed int pr;
    unsigned int prz, i, j, j_incr, zakres;
    unsigned int znalezione, reszta, flaga;
    prz = atoi(argv[1]);

    znalezione = 1; //2 to liczba pierwsza (1 NIE jest liczbą pierwszą bo pierwsze muszą by wieksze od 1)


    for (i=3; i <= prz; i=i+2) {
        flaga = 0;
        j_incr = 1;
        zakres = (unsigned int) sqrt(i) + 1;
        for (j=2; j<zakres; j = j + j_incr) {
            reszta = i%j;
            if (j == 3 ) j_incr = 2;
            if (reszta == 0) {
                flaga++;
                break;
            }
        }

        if (flaga == 0) {
            znalezione++;
        }
   }

    printf("W przedziale do %i znaleziono %i liczb pierwszych\n", prz, znalezione);
    return 0;
    
}

File: liczbyp.py
from math import sqrt
import sys

prz = int(sys.argv[1])
znalezione = 1  # 2 to liczba pierwsza (1 NIE jest!)

for i in range(3, prz+1, 2):
    flaga = 0
    zakres = int(sqrt(i)+1)
    dzielniki = [i for i in range(2, zakres) if i < 3 or i % 2 != 0]
    for j in dzielniki:
        reszta = i % j
        if reszta == 0:
            flaga += 1
            break
    if flaga == 0:
        znalezione += 1

print("W przedziale do %s znaleziono %s liczb pierwszych" % (prz, znalezione))
